### PR TITLE
(#77) Return file path from Take Screenshot

### PR DESCRIPTION
**Pull request by Harm10**
The change looks fairly simple. As I am a bit new on PR's through GitHub perhaps my way of creating could be improved?

### CODE REVIEW

**Comment by robinmatz** [comment]
Hey, the changes look good to me. Even though it's a simple change, I think it would be best if you added a unit test.

For this, it would be ok if you modifies the exiting test under `utests/x3270/test_screenshot.py`

Let me know if you need further help

**Comment by Harm10** [comment]
I do not have experience with utests but I will have a look!
Are these the workflows awaiting approval?

**Comment by Harm10** [comment]
Forgive my ignorance :-) : but how should I execute this test?
If I do python test_screenshot.py in a terminal window I get
ModuleNotFoundError: No module named 'pytest_mock'

**Comment by Harm10** [comment]
I went through the Development setup documentation and could use invoke utest to get:

```
utest\Mainframe3270\test_run_on_failure.py ...                           [  2%]
utest\py3270\test_command.py .......                                     [  8%]
utest\py3270\test_emulator.py ................                           [ 23%]
utest\py3270\test_status.py ..                                           [ 24%]
utest\x3270\test__init__.py ....                                         [ 28%]
utest\x3270\test_assertions.py ........................................  [ 63%]
utest\x3270\test_commands.py .........                                   [ 71%]
utest\x3270\test_connection.py ......                                    [ 76%]
utest\x3270\test_read.py .....                                           [ 81%]
utest\x3270\test_screenshot.py ...                                       [ 84%]
utest\x3270\test_wait_and_timeout.py ..............                      [ 96%]
utest\x3270\test_write.py ....                                           [100%]
```

So I am still a bit at a loss what is expected?

**Comment by Harm10** [comment]
I think I get the logic of this. I managed to update the test to check for the correct filepath returned. I will also look at the atest as these also give FAILures.

**Comment by Harm10** [comment]
I did an additional commit to my branch for utest change, invoke lint and atest changes. It looks like they are added immediately to  the PR?
@robinmatz Can you confirm?

BTW The atest fails on 2 tests looking for a specific text on screen of PUB400 if the user logging in has a different language (like me). So perhaps you should validate something else (like a menu sequence number)?

File: Mainframe3270/x3270.py [hunk 1]
```diff
@@ -207,16 +209,18 @@ def set_screenshot_folder(self, path: str) -> None:
...
         The Screenshot is printed in a iframe log, with the values of height=410 and width=670, you
         can change this values passing them to the keyword.
 
+        The full file path is returned.
+
         Example:
-            | Take Screenshot |
-            | Take Screenshot | height=500 | width=700 |
+            | ${filepath} | Take Screenshot |
+            | ${filepath} | Take Screenshot | height=500 | width=700 |
```
Comment (robinmatz) [inline Mainframe3270/x3270.py:223]: @Harm10 , I think it's also good to have one example where the screenshot is not returned

File: utest/x3270/test_screenshot.py [hunk 2]
```diff
@@ -32,10 +32,12 @@ def test_take_screenshot(mocker: MockerFixture, under_test: x3270):
...
-    under_test.take_screenshot(500, 500)
+    filepath = under_test.take_screenshot(500, 500)
 
     logger.write.assert_called_with(
         '<iframe src="./screenshot_1000.html" height="500" width="500"></iframe>',
         level="INFO",
         html=True,
     )
+    
+    assert filepath == ".\\screenshot_1000.html"
```
Comment (robinmatz) [inline utest/x3270/test_screenshot.py:43]: Even though, I hate logic in tests, I think the easiest solution to this problem is

```python
if os.name == "nt":
        assert filepath == r".\screenshot_1000.html"
    else:
        assert filepath == "./screenshot_1000.html"
```

**Comment by Harm10** [comment]
@robinmatz I have resolved both your suggestions. This is also new to me. It looks like your changes are now part of this PR. They are not part of my branch so probably I need to do something additional. I will investigate.

I noticed one of your changes had to do with the ubuntu 3.7 failing. Should I be able to redo this test? 
The other failing test has to do with ubuntu failing on python 3.10. Could you have a look?

**Comment by Harm10** [comment]
I was mistaken about resolving your suggestions made them part of this PR. I will do them manually in my branch. 
Is there a way to do this automatically?

**Comment by robinmatz** [comment]
> I was mistaken about resolving your suggestions made them part of this PR. I will do them manually in my branch. Is there a way to do this automatically?

I reviewed your changes just now. As you correctly noticed, the suggestions I made were only part of the conversation of this PR. The way GitHub displays these suggestions might be a bit confusing at first.

As for the failing pipeline, you need not to worry about this, since the failures are not related to your changes. I will look into these on another occasion.

**Review by robinmatz — APPROVED** [review]

**Pull request merged**